Automattic/vip-cli
# Improve login + logout

## Description

Adds server-side logout from the `vip logout` command. This action deactivates the token and prevents further use.

Also prevents "unknown command" errors when running `vip login`, effectively adding `vip login` as a valid command (though due to how it's architected, it still doesn't show in `vip -h`). Previously, the only way to login was to run some non-login command while you were logged out, which is unexpected. `login` is a standard command across CLI tools.

## Steps to Test

1. Check out PR.
1. Run `npm run build`
1. Run `./dist/bin/vip.js logout` - start fresh
1. Run `./dist/bin/vip.js login`
2. When generating the token on the Dashboard, save it for later use
3. Run regular commands to ensure things work - `./dist/bin/vip-app-list.js` for example
4. Via curl, or Insomnia, or Postman, make an HTTP request to the API with your token in `Authorization: Bearer $TOKEN` header ... should succeed
5. Logout again .... `./dist/bin/vip.js logout`
6. In the other program, repeat the request and you should now receive a 401 indicating your token is no longer valid
